src/ode_solver/low_storage_runge_kutta_ode_solver.cpp

```diff
@@ -0,0 +1,188 @@
+        // rhs = dt * f(S1)
+        rhs = storage_register_1;
+    }
+   // std::abort();
+    // storage_register_2 = (storage_register_2 + delta[m] * storage_register_1 + delta[m+1] * storage_register_3) / sum_delta;
+    if (this->butcher_tableau->get_b_hat(1) == 0){
+        for (int i = 0; i < num_delta; i++){ //change n_rk_stages+2 to num_delta on this line
+        sum_delta = sum_delta + this->butcher_tableau->get_delta(i);
+        }
+        storage_register_2.add(this->butcher_tableau->get_delta(n_rk_stages), storage_register_1);
+        storage_register_2.add(this->butcher_tableau->get_delta(n_rk_stages+1), storage_register_3);
+        storage_register_2 /= sum_delta;
+        // u_hat = s2
+    }
+
+
+    // need to calculate rhs of s1
+    else if (this->butcher_tableau->get_b_hat(n_rk_stages) != 0){
+        this->dg->solution = rhs;
+        this->dg->assemble_residual();
+        this->dg->apply_inverse_global_mass_matrix(this->dg->right_hand_side, rhs);
+        rhs *= dt;
+        storage_register_4.add(this->butcher_tableau->get_b_hat(n_rk_stages), rhs);
+        //this->pcout << " b_hat_fsal " << this->butcher_tableau->get_b_hat(n_rk_stages);
+
+    }
+    this->dg->solution = storage_register_1;
+    
+    double global_size = dealii::Utilities::MPI::sum(storage_register_1.local_size(), this->mpi_communicator);
+    
+    if (this->butcher_tableau->get_b_hat(1) == 0){
+        // loop sums elements at each mpi processor
+        for (dealii::LinearAlgebra::distributed::Vector<double>::size_type i = 0; i < storage_register_1.local_size(); ++i) {
+            error = storage_register_1.local_element(i) - storage_register_2.local_element(i);
+            w = w + pow(error / (atol + rtol * std::max(std::abs(storage_register_1.local_element(i)), std::abs(storage_register_2.local_element(i)))), 2);
+        }
+    }
+    else if (this->butcher_tableau->get_b_hat(1) != 0){
+        // loop sums elements at each mpi processor
+        for (dealii::LinearAlgebra::distributed::Vector<double>::size_type i = 0; i < storage_register_1.local_size(); ++i) {
+            error = storage_register_1.local_element(i) - storage_register_4.local_element(i);
+            w = w + pow(error / (atol + rtol * std::max(std::abs(storage_register_1.local_element(i)), std::abs(storage_register_4.local_element(i)))), 2);
+        }
+    }
+    this->pcout << std::endl;
+
+    // sum over all elements
+    w = dealii::Utilities::MPI::sum(w, this->mpi_communicator);
+    
+    w = pow(w / global_size, 0.5);
+
+    this->pcout << std::endl;
+    // std::cout << "w " << w;
+
+    ++(this->current_iteration);
+    this->current_time += dt;
+    this->pcout << " Time is: " << this->current_time <<std::endl;
+    this->pcout << "dt" << dt;
+}
+
+
+template <int dim, typename real, int n_rk_stages, typename MeshType> 
+double LowStorageRungeKuttaODESolver<dim,real,n_rk_stages, MeshType>::err_time_step (real dt, const bool pseudotime)
```

Comment:
Can we use a name similar to the ones used in flow solver for consistency? Maybe `get_automatic_error_adaptive_step_size`.